dash/_pages.py

```diff
@@ -358,7 +358,12 @@ def register_page(
     # Sort by order and module, then by module
     for page in sorted(
         PAGE_REGISTRY.values(),
-        key=lambda i: (str(i.get("order", i["module"])), i["module"]),
+        key=lambda i: (
+            "order" not in i,  # False (order given) sorts before True
+            i["order"] if isinstance(i.get("order"), (int, float)) else float("inf"),
+            str(i.get("order")),
```

Comment:
OK I see what was happening, the entry in `page_registry` always has `order`, it's just `None` if it wasn't provided - so `"order" not in i` was always `False`.
```suggestion
            i["order"] is None,  # False (order given) sorts before True
            i["order"] if isinstance(i["order"], (int, float)) else float("inf"),
            str(i["order"]),
```